authentication/oauth_test.go

```diff
@@ -21,3 +21,123 @@ func TestOAuthLoginWithPassword(t *testing.T) {
 	assert.NotEmpty(t, tokenSet.IDToken)
 	assert.Equal(t, "Bearer", tokenSet.TokenType)
 }
+
+func TestLoginWithAuthCode(t *testing.T) {
+	t.Run("Should require client_secret", func(t *testing.T) {
+		_, err := authAPI.OAuth.LoginWithAuthCode(context.Background(), oauth.LoginWithAuthCodeRequest{
+			Code: "my-code",
+		})
+
+		assert.Error(t, err, "client_secret is required but not provided")
+	})
+
+	t.Run("Should throw for an invalid code", func(t *testing.T) {
+		configureHTTPTestRecordings(t)
+
+		_, err := authAPI.OAuth.LoginWithAuthCode(context.Background(), oauth.LoginWithAuthCodeRequest{
+			ClientAuthentication: oauth.ClientAuthentication{
+				ClientSecret: clientSecret,
+			},
+			Code: "my-invalid-code",
+		})
+
+		assert.Error(t, err, "Invalid authorization code")
+	})
+
+	t.Run("Should return tokens", func(t *testing.T) {
+		configureHTTPTestRecordings(t)
+
+		tokenSet, err := authAPI.OAuth.LoginWithAuthCode(context.Background(), oauth.LoginWithAuthCodeRequest{
+			ClientAuthentication: oauth.ClientAuthentication{
+				ClientSecret: clientSecret,
+			},
+			Code: "my-code",
+		})
+
+		assert.NoError(t, err)
+		assert.NotEmpty(t, tokenSet.AccessToken)
+		assert.NotEmpty(t, tokenSet.IDToken)
+		assert.Equal(t, "Bearer", tokenSet.TokenType)
+	})
+}
+
+func TestLoginWithAuthCodeWithPKCE(t *testing.T) {
```

Comment:
Should this include a "Should throw for an invalid code" test case?

Reply:
Will add